test/s3path.jl

```diff
@@ -334,4 +334,13 @@ end
     end
 end
 
+@testset "JSON roundtripping" begin
+    json_path = S3Path("s3://$(bucket_name)/test_json"; config=aws)
+    my_dict = Dict("key" => "value", "key2" => 5.0)
+    s3_put(aws, bucket_name, "test_json", JSON3.write(my_dict), "application/json")
+    json_bytes = read(json_path)
+    @test JSON3.read(json_bytes, Dict) == my_dict
+    rm(json_path)
```

Comment:
since we are in a `@testset`, this line should get called even if a previous line throws. Besides, we nuke the bucket at the end anyway.

Reply:
If an exception is raised then the `@testset` will be aborted early.

Reply:
oh really? good to know!